io-engine/src/bdev/nexus/nexus_io.rs

```diff
@@ -270,10 +281,37 @@ impl<'n> NexusBio<'n> {
             self.resubmit();
         } else {
             error!("{self:?}: failing nexus I/O: all child I/Os failed");
+
+            unsafe {
+                self.nexus_mut().get_unchecked_mut().last_error = status;
+            }
+
             self.fail();
         }
     }
 
+    /// Fails the current I/O with a generic internal error. If the nexus
+    /// already had a last child error, it fails with it.
+    fn fail(&self) {
```

Comment:
Could this be problematic for thin provisioning cases @tiagolobocastro @dsavitskiy . Say, first a replica fails with some nvme error, and then a replica fails the same IO with ENOSPC. This function will make the nexus IO fail with the nvme error because that happened first.